src/ui/molecules/header/Header.tsx

```diff
@@ -1,30 +1,119 @@
-import React from 'react'
-import './header.scss'
-import { ThemeSwitcher } from 'ui/atoms/theme/ThemeSwitcher'
-import type { DeepReadonly } from 'superTypes'
-import { useBreakpoint } from 'hook/useBreakpoint'
-import type { Breakpoints } from 'hook/useBreakpoint'
+import React, { useCallback, useState } from 'react'
+import type { DeepReadonly, UseState } from 'superTypes'
 import classNames from 'classnames'
+import { useMediaType } from 'hook/useMediaType'
+import { ThemeSwitcher as Switcher } from 'ui/atoms/theme/ThemeSwitcher'
+import { Icon } from 'ui/atoms/icon/Icon'
+import './header.scss'
 
 export type HeaderProps = {
   /**
    * Element positioned first in the reading flow.
    */
   readonly firstElement: JSX.Element
+  /**
+   * The list of navigable links that make up the menu.
+   */
+  readonly navigationMenu?: JSX.Element[]
 }
 
+const BurgerMenu = ({
+  isOpened,
+  onToggle
+}: DeepReadonly<{ isOpened: boolean; onToggle: () => void }>): JSX.Element => (
```

Comment:
IMHO, it's better to name explicitely the props type.

Reply:
Isn't "isOpen" a better name than "isOpened"?

Reply:
I agree with you but last time we agreed with @ccamel that it was not disturbing to not name them when it was internal to the component.

Reply:
as we decided, let's go for `isOpen` and `isBurgerMenuOpen` !